### PR TITLE
v1: backport workflow/label changes for semver+label

### PR DESCRIPTION
**Pull request by EmilienM**
Backport of the following commits from master:

- [CI] Drop branch filters for 'Ensure Labels' and 'CodeQL' jobs

So they can run on stable branches too.

- Instrument backporting to v1

Add the changes to semver & labels, useful for `v1` branch.
The bits around backports are removed in another commit (see below).

- chore: Fix labels

- Remove backport workflow & label

We don't want the backport workflow (and its labels) to exist in v1.

### CODE REVIEW

**Comment by EmilienM** [comment]
> Once the backporting automation is working propertly, I'd like to let it deal with the backports. WDYT?

Do you mean you want to do the backports that I manually did, via the new automated system?
Not all backports were clean, so I'm not sure it'll work. But I'm fine either way.

File: .github/workflows/ensure-labels.yaml [hunk 1]
```diff
@@ -1,18 +1,16 @@
-name: Ensure labels
+name: Apply labels in .github/labels.yaml
 on:
   push:
-    branches:
-      - master
```
Comment (mandre) [inline .github/workflows/ensure-labels.yaml:5]: As Pierre pointed out the other day, this workflow syncs the repository labels from what is defined in the `.github/labels.yaml` file. If we were to run this from the v1 branch, we would delete the `backport-v1` label each time we had a PR triggering this workflow in the `v1` branch. This was my mistake and we should revert it.

File: README.md [hunk 2]
```diff
@@ -1,13 +1,11 @@
...
 
-Gophercloud is an OpenStack Go SDK.
+[Reference documentation](http://godoc.org/github.com/gophercloud/gophercloud)
 
-## Useful links
+Gophercloud is a Go SDK for OpenStack.
 
-* [Reference documentation](http://godoc.org/github.com/gophercloud/gophercloud)
-* [Effective Go](https://golang.org/doc/effective_go.html)
+This is the development branch of Gophercloud; stable releases are cut from the branch `v1`.
```
Comment (mandre) [inline README.md:8]: This should be changed for the `v1` branch.

**Review by mandre — CHANGES_REQUESTED** [review]

File: .github/workflows/ensure-labels.yaml (hunk 1 above)
Comment (mandre) [reply .github/workflows/ensure-labels.yaml:5]: In fact, we should even delete this workflow from the v1 branch.

Comment (EmilienM) [reply .github/workflows/ensure-labels.yaml:5]: I'm a bit confused by what we want now, I'll let @pierreprinetti comment; you seem to have a stronger opinion on what we should be doing wrt these workflows in v1 branch. Let me know if you need help with that PR.

Comment (mandre) [reply .github/workflows/ensure-labels.yaml:5]: Closing in favor of https://github.com/gophercloud/gophercloud/pull/2683

**Comment by mandre** [comment]
Closing in favor of https://github.com/gophercloud/gophercloud/pull/2683.

**Pull request closed without merging**